src/mission_ui.cpp

```diff
@@ -17,9 +17,216 @@
+        if( switch_tab == mission_ui_tab_enum::ACTIVE ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
+        }
+        if( ImGui::BeginTabItem( _( "ACTIVE" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::ACTIVE;
+            umissions = get_avatar().get_active_missions();
+            ImGui::EndTabItem();
+        }
+        flags = ImGuiTabItemFlags_None;
+        if( switch_tab == mission_ui_tab_enum::COMPLETED ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
+        }
+        if( ImGui::BeginTabItem( _( "COMPLETED" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::COMPLETED;
+            umissions = get_avatar().get_completed_missions();
+            ImGui::EndTabItem();
+        }
+        flags = ImGuiTabItemFlags_None;
+        if( switch_tab == mission_ui_tab_enum::FAILED ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
+        }
+        if( ImGui::BeginTabItem( _( "FAILED" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::FAILED;
+            umissions = get_avatar().get_failed_missions();
+            ImGui::EndTabItem();
+        }
+        ImGui::Separator();
+        ImGui::EndTabBar();
+    }
+
+    if( selected_mission < 0 ) {
+        selected_mission = 0;
+    }
+
+    if( selected_mission > umissions.size() - 1 ) {
+        selected_mission = umissions.size() - 1;
+    }
+
+    if( umissions.empty() ) {
+        ImGui::Text( _( "No missions!" ) );
+        return;
+    }
+
+    // All window sizes are currently static values until I've finished building the thing.
+    if( ImGui::BeginTable( "##MISSION_TABLE", 2, ImGuiTableFlags_None, ImVec2( 900.0f, 500.f ) ) ) {
+        ImGui::TableSetupColumn( _( "Missions" ), ImGuiTableColumnFlags_WidthFixed, 300.0f );
+        ImGui::TableSetupColumn( _( "Description" ), ImGuiTableColumnFlags_WidthFixed, 600.0f );
+        ImGui::TableHeadersRow();
+        ImGui::TableNextColumn();
+        draw_mission_names( umissions, selected_mission );
+        ImGui::TableNextColumn();
+        draw_selected_description( umissions, selected_mission );
+        ImGui::EndTable();
+    }
+}
+
+void mission_ui_impl::draw_mission_names( std::vector<mission *> missions, int &selected_mission )
+{
+    const int num_missions = missions.size();
+    if( ImGui::BeginListBox( "##LISTBOX", ImVec2( 300.0f, 300.0f ) ) ) {
+        for( int i = 0; i < num_missions; i++ ) {
+            const bool is_selected = ( selected_mission == i );
+            if( ImGui::Selectable( missions[i]->name().c_str(), is_selected ) ) {
+                selected_mission = i;
+            }
+
+            if( is_selected ) {
+                ImGui::SetScrollHereY();
+                ImGui::SetItemDefaultFocus();
+            }
```

Comment:
```suggestion
            ImGui::PushID( i );
            if( ImGui::Selectable( missions[i]->name().c_str(), is_selected ) ) {
                selected_mission = i;
            }

            if( is_selected ) {
                ImGui::SetScrollHereY();
                ImGui::SetItemDefaultFocus();
            }
            ImGui::PopID();
```

If you had two missions with the same name, their entries would be synchronized because they'd have the same id. I'm not sure if that's ever supposed to happen, but like this you can make sure they're always their own separate entries.
(I'm not actually sure at what point calling `PopID` is safe, but putting it down there definitely isn't wrong.)